src/RA_Dlg_Achievement.cpp

```diff
@@ -227,22 +223,19 @@ BOOL LocalValidateAchievementsBeforeCommit(int nLbxItems[1])
             return FALSE;
         }
 
-        char sIllegalChars[] = { '&', ':' };
+        std::array<char, 2> sIllegalChars{ '&', ':' };
 
-        const size_t nNumIllegalChars = sizeof(sIllegalChars) / sizeof(sIllegalChars[0]);
-        for (size_t i = 0; i < nNumIllegalChars; ++i)
+        for (auto& c : sIllegalChars)
         {
-            char cNextChar = sIllegalChars[i];
-
-            if (strchr(Ach.Title().c_str(), cNextChar) != nullptr)
+            if (Ach.Title().find_first_of(c) != std::string::npos)
             {
-                sprintf_s(buffer, 2048, "Achievement title contains an illegal character: '%c'\nPlease remove and try again", cNextChar);
+                sprintf_s(buffer, 2048, "Achievement title contains an illegal character: '%c'\nPlease remove and try again", c);
                 MessageBox(nullptr, NativeStr(buffer).c_str(), TEXT("Error!"), MB_OK);
                 return FALSE;
             }
-            if (strchr(Ach.Description().c_str(), cNextChar) != nullptr)
+            if (Ach.Description().find_first_of(c) != std::string::npos)
             {
-                sprintf_s(buffer, 2048, "Achievement description contains an illegal character: '%c'\nPlease remove and try again", cNextChar);
+                sprintf_s(buffer, 2048, "Achievement description contains an illegal character: '%c'\nPlease remove and try again", c);
```

Comment:
Forgot to leave a comment in code here but this was another case of spectre vulnerability. There's still 8 in the code but thought I'd just fix them later.